src/zeek.bif

```diff
@@ -2961,26 +2961,32 @@ function bytestring_to_count%(s: string, is_le: bool &default=F%): count
 	static const bool host_bigendian = false;
 #endif
 	const u_char *p = s->Bytes();
-	unsigned int i;
+	unsigned int i = 0;
 
-	switch ( s->Len() ) {
-	case sizeof(uint8_t):
+	if ( s->Len() == 0 )
+		{
+		return zeek::val_mgr->Count(0);
+		}
+	if ( s->Len() == sizeof(uint8_t) )
 		{
 		uint8_t value = 0;
 		memcpy(&value, p, sizeof(uint8_t));
 		return zeek::val_mgr->Count(value);
 		}
-
-	case sizeof(uint16_t):
+	else if ( s->Len() <= sizeof(uint16_t) )
```

Comment:
This is always going to be a `sizeof(uint16_t)`, no? (Because otherwise the above if-clause will kick in?)

Reply:
Good point. I'll just make that `==`.

Reply:
That also means that block doesn't need to change at all. I'll remove all of those changes.